tower/src/buffer/layer.rs

```diff
@@ -9,12 +9,13 @@ use tower_service::Service;
 /// which means that this layer can only be used on the Tokio runtime.
 ///
 /// See the module documentation for more details.
-pub struct BufferLayer<Request> {
+pub struct BufferLayer<Request, E2> {
     bound: usize,
     _p: PhantomData<fn(Request)>,
+    _e: PhantomData<E2>,
```

Comment:
also i would probably just have one `PhantomData` for everything, 
```suggestion
    _p: PhantomData<fn(Request, E2)>,
```
having two phantom fields just adds noise IMO